models/concerns/validate_data.rb

```diff
@@ -139,6 +139,14 @@ def check_services(schedule)
   def check_relation_consistent_timewindows
     unconsistent_relation_timewindows = []
     @hash[:relations].each{ |relation|
+      unless relation[:linked_ids].uniq == relation[:linked_ids]
+        raise OptimizerWrapper::DiscordantProblemError.new('Same service ID was provided several times in one relation.')
```

Comment:
it can be service or shipment so, it is better as follows:

Same ID is provided multiple times in the same relation